clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -1188,6 +1206,14 @@ boolean isEmpty() {
                     inFlightAcknowledgements.isEmpty();
         }
 
+        /**
+         * Resets the timer with the new deadline and resets the RequestState.
```

Comment:
nit: Probably "new timeout" rather than "new deadline" given that you've replaced the concept of deadline with a timeout.